src/analytics.coffee

```diff
@@ -6,17 +6,41 @@ define [
 ], (Settings, Session, PluginsManager, ActionsManager) ->
   class Analytics
     constructor: ->
-      plugins_manager = new PluginsManager
+      @plugins_manager = new PluginsManager
+      @session = new Session(@plugins_manager)
+      @actions = null
 
-      new Session(plugins_manager).run().then (session) =>
-        @actions = new ActionsManager(session, plugins_manager).run()
+      if Settings.window[Settings.command_queue_name].q?.length > 0
+        @_handle_session()
+      else
+        ###
+        Replaces our queue appending global function with one which immediately
+        runs commands when the commands array is empty or undefined.
+        Intended for programmatic usage and/or lazy command declaration.
+        ###
+        Settings.window[Settings.command_queue_name] = =>
+          (Settings.window[Settings.command_queue_name].q ||= []).push(arguments)
+          @_handle_session()
```

Comment:
wouldn't it be better to make this a `_replace_queue` function and call it in the two places we need it?

Reply:
I think the extraction here is unnecessary, we know the code's purpose. Despite that I tried to extract it and we would have to update a lot of specs outside of analytics_spec.coffee as well and that makes the change even more complex.

Reply:
alright understood, we can keep it as it is then